File: src/main/java/org/revo/track/Domain/Location.java
package org.revo.track.Domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Document
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Location {
    @Id
    private String id;
    private String trackerId;
    private Date date=new Date();
    private Double lat;
    private Double lng;
}

File: src/main/resources/ui-static/src/app/Domain/Location.ts
export class Location {

  constructor(id: string, trackerId: string, date: string, lat: number, lng: number) {
    this.id = id;
    this.trackerId = trackerId;
    this.date = date;
    this.lat = lat;
    this.lng = lng;
  }

  id: string;
  trackerId: string;
  date: string;
  lat: number;
  lng: number;
}
